STEER/STEERBase/AliMagFast.h

```diff
@@ -71,9 +102,31 @@ inline float AliMagFast::CalcPol(const float* cf, float x,float y, float z) cons
     x*(cf[1] + x*(cf[4] + x*cf[10] + y*cf[11] + z*cf[12]) + y*(cf[5]+z*cf[14]) ) +
     y*(cf[2] + y*(cf[7] + x*cf[13] + y*cf[16] + z*cf[17]) + z*(cf[8]) ) +
     z*(cf[3] + z*(cf[9] + x*cf[15] + y*cf[18] + z*cf[19]) + x*(cf[6]) );
-    
+
   return val;
 }
 
+inline Bool_t AliMagFast::QuickSearch(const SegmentSearch_t ss, const float z, UShort_t &id) const
+{
+  const int index = floor((z - ss.offset) * ss.factor);
+  if (index > ss.nDivision || index<0) return kFALSE;
```

Comment:
Related to previous: If the nDivision is indeed the number of divisions, should not you use here
````
if (index >= ss.nDivision || index<0) return kFALSE;
````
Also, to avoid ushort - int comparison, I would use int for nDivision

Reply:
Sorry, wait a moment

Reply:
So, do you keep here ````index > ss.nDivision````?

Reply:
index can be any integer including negative number.
So `index < 0` is necessary. `(so index > -1 && index < 0)` is alway false.

Reply:
OK, local build succeeded!

Reply:
@syamasak I mean is index==ss.nDivision valid? From the following code I would say no.

Reply:
What a basic mistake... 
// Why humans should take care such kind of mistake? (because the compiler is not smart enough.)

Any other bug or suggestion? I'll make the PR for the data.

Reply:
@syamasak so, if it should be ````index >= ss.nDivision````, would you fix it? 
The rest looks fine.

Reply:
@shahor02 Oh, the fix has not staged correctly. I push it now.

Reply:
@shahor02 this version seems ok. Thank you!